File: server/models/Customer.js
const mongoose = require('mongoose');

const CustomerSchema = new mongoose.Schema(
  {
    MaKH: {
      type: String,
      required: true,
      unique: true,
    },
    TenKH: {
      type: String,
      required: true,
    },
    DiaChiKhachHang: {
      type: String,
      default: '',
    },
    SDTKhachHang: {
      type: String,
      default: '',
    },
  },
  { timestamps: true, collection: 'customers' }
);

module.exports = mongoose.model('Customer', CustomerSchema);

File: lib/core/models/customer.dart
class Customer {
  final String id;
  final String maKH;
  final String tenKH;
  final String diaChi;
  final String sdt;

  Customer({
    required this.id,
    required this.maKH,
    required this.tenKH,
    required this.diaChi,
    required this.sdt,
  });

  factory Customer.fromJson(Map<String, dynamic> json) {
    return Customer(
      id: json['_id'] ?? '',
      maKH: json['MaKH'] ?? '',
      tenKH: json['TenKH'] ?? 'Khách lẻ',
      diaChi: json['DiaChiKhachHang'] ?? '',
      sdt: json['SDTKhachHang'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'MaKH': maKH,
      'TenKH': tenKH,
      'DiaChiKhachHang': diaChi,
      'SDTKhachHang': sdt,
    };
  }
}
